src/components/App.js

```diff
@@ -1,15 +1,33 @@
+import React from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
-// eslint-disable-next-line no-unused-vars
-import Calculate from '../logic/calculate';
-
-function App() {
-  return (
-    <>
-      <Display result="0" />
-      <ButtonPanel />
-    </>
-  );
-}
+import calculate from '../logic/calculate';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      total: null,
+      next: null,
+      // eslint-disable-next-line react/no-unused-state
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
 
-export default App;
+  handleClick(buttonName) {
+    const data = this.state;
+    this.setState(calculate(data, buttonName));
+  }
+
+  render() {
+    const data = this.state;
+    return (
+      <>
+        <Display result={data.next || data.total || '0'} />
+        <ButtonPanel clickHandler={this.handleClick} />
```

Comment:
- [ ] Use Destructuring 

```JavaScript
   const { next, total } = this.state; // Destructuring 
   return (
      <>
        <Display result={ next || total || '0'} /> // I called next and total without having to call data.
        <ButtonPanel clickHandler={this.handleClick} />
      </>
```

Reply:
This last one is fine, I just noticed it.

The first request isn't making much sense to me, how can I add the function inside the constructor?
The code given isn't the solution, and never do what you're asking, even though I'm trying to find how to implement the function inside of it, I can not find a clear answer.

Can you elaborate on that one?